File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>

using namespace std;

string intToBin(int n) {
    string s;
    while (n > 0) {
        s.append(1, n % 2 + '0');
        n /= 2;
    }
    return s;
}

vector<int> solution(string s) {
    int cnt = 0;
    int zero = 0;
    while (s.length() != 1) {
        cnt++;
        int tmp = 0;
        for (auto c : s) (c == '0') ? zero++ : tmp++;
        s = intToBin(tmp);
    }
    return {cnt, zero};
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int transCount = 0, removeCount = 0;
        for (; s.length() > 1; transCount++) {
            int tmp = s.length();
            s = s.replace("0", "");
            removeCount += tmp - s.length();
            s = Integer.toBinaryString(s.length());
        }
        return new int[]{transCount, removeCount};
    }
}